data/src/client.rs

```diff
@@ -1270,8 +1282,27 @@ impl Client {
         }
     }
 
+    // TODO allow configuring the "sorting method"
+    // this function sorts channels together which have similar names when the chantype prefix
+    // (sometimes multipled) is removed
+    // e.g. '#chat', '##chat-offtopic' and '&chat-local' all get sorted together instead of in
+    // wildly different places.
+    fn compare_channels(&self, a: &str, b: &str) -> Ordering {
+        let (Some(a_chantype), Some(b_chantype)) = (a.chars().nth(0), b.chars().nth(0)) else {
+            return a.cmp(b);
+        };
+
+        if [a_chantype, b_chantype].iter().all(|c| self.chantypes().contains(c)) {
+            let ord = a.trim_start_matches(a_chantype).cmp(b.trim_start_matches(b_chantype));
+            if ord != Ordering::Equal {
+                return ord;
+            }
+        }
+        a.cmp(b)
+    }
```

Comment:
I think we're long overdue for a `Channel` type in our data crate. We're exposing a ton of the complexities of parsing / dealing w/ channel prefix vs names across the codebase instead of encapsulating it within a `channel` module / type.

I'm not sure we should block this work on it as it'd be a much bigger refactor, but I think it'd be really nice to clean this up along w/ dealing w/ channels and "targets" (channel vs query "nick") throughout the codebase instead of just `String` like we've been doing.

Reply:
one issue is that Ord (as in hashmap) and Ord as in channel sorting are (and should be!) different

we really want a locale-dependent sort on channels and i don't think this is suitable for `Ord`, which the BTreeMap is based on (for an example of this see my [gamja PR](https://codeberg.org/emersion/gamja/pulls/4))

I wasn't entirely sure on how to rectify this so i decided to not work on it in this PR... so maybe we should push it to another PR?

Reply:
I opened #626 to track this